modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js

```diff
@@ -123,14 +159,36 @@ export default function DatePicker({
 	 * Creates the input mask and update it whenever the format changes
 	 */
 	useEffect(() => {
-		const mask = [...momentFormat].map((char) =>
-			LETTER_REGEX.test(char) ? DIGIT_REGEX : char
-		);
+		const mask = [];
+		[...momentFormat].forEach((char) => {
+			if (char === 'A' || char === 'a') {
+				mask.push(/[AP]/i);
+				mask.push(/M/i);
```

Comment:
Hi @rebsilva, I didn't get what `/[AP]/` and `/M/i` means. Could you clarify that by creating a constant, please?

Reply:
`/[AP]/i` means that it will accept only characters like `A, P, a or p`, and `/M/i` means it will only accept the `M or m` character. I will create constants for them, thanks!